Use `schema_name` consistently

tijmenb opened the pull request:
The content item attribute `format` is deprecated.

alphagov/govuk-content-schemas#444 attempts to remove `format` from the frontend, but the tests for this application fail without `format` in the examples. This is because the tests are testing kind of the wrong thing.

For example:

```
assert_equal schema_item['format'], presented_item.format
```

Checks the string in `presented_item.format` which actually points to `schema_item['schema_name']`. Because they're `format` and `schema_name` are identical in the examples this worked. Now that we've removed `format` it doesn't anymore.

https://trello.com/c/C8A9bnuO/344-work-on-making-the-document-type-concept-clearer

andrewgarner commented on app/presenters/national_applicability.rb:
```diff
@@ -23,8 +23,8 @@ def applies_to
-  def translated_format_name(count)
-    I18n.t("content_item.format.#{format}", count: count).downcase
+  def translated_schema_name(count)
+    I18n.t("content_item.format.#{schema_name}", count: count).downcase
```
Should we update the locale key to `content_item.schemas.x`?

tijmenb replied:
Yes, probably. It does feel a little more risky though, and afaik they were imported from Whitehall originally so I'm a bit hesitant to touch them as part of this PR.

andrewgarner replied:
Fair enough. It might be worth leaving a comment in the code so that future intend is not forgotten.

cbaines commented:
It's only useful for the frontend to have a concept of the schema_name if it needs to do validation, as that is what it should use as the filename of the schema. I think its best to use schema_name for this (its intended purpose) and no other purpose, as this will help avoid confusion. Is using the schemas something that is planned for the government-frontend, as I can't see any use of them within the app currently?

tijmenb commented:
@cbaines this app determines the layout of the page depending on `schema_name`. I think it's worth having a discussion about that (should it be `document_type`, or even a new `template_name`?). Probably out of scope for this PR though.

tijmenb commented:
I've updated the commit to only change the bare minimum to get the tests on `govuk-content-schemas` passing. Removed the changes to make it use the `schema_name` consistently, because there's still translation keys called `content_item.format` (https://github.com/alphagov/government-frontend/pull/202#discussion_r90010580) and I don't have enough confidence in the tests to also fix those.

The original branch is here:

https://github.com/alphagov/government-frontend/compare/use-schema-name-consistently

andrewgarner commented:
👍
